### PR TITLE
Allow wires to be passed as final positional argument

### PR DESCRIPTION
**Pull request by albi3ro**
Currently, the initialization of `Observable` has logic that allows the wires to be passed as the final positional argument.

I move this logic to `Operator` to make it more general.  This also allows us to get rid of a specific `__init__` function for `Observable`'s.

I also tidied two other chunks in the `Operator` initialization function while I was there.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -144,6 +144,15 @@
 
 <h3>Improvements</h3>
 
+* Wires can be passed as the final argument to an `Operator`, instead of requiring
+  the wires to be explicitly specified with keyword `wires`. This functionality already
+  existed for `Observable`'s, but now extends to all `Operator`'s.
+
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:150]:
```suggestion
  [(#2432)](https://github.com/PennyLaneAI/pennylane/pull/2432)

```

File: pennylane/operation.py [hunk 2]
```diff
@@ -878,30 +878,34 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
...
         self.queue_idx = None  #: int, None: index of the Operator in the circuit queue, or None if not in a queue
 
+        wires_from_args = False
         if wires is None:
-            raise ValueError(f"Must specify the wires that {self.name} acts on")
+            try:
+                wires = params[-1]
+                params = params[:-1]
+                wires_from_args = True
+            except IndexError as err:
```
Comment (Jaybsoni) [inline pennylane/operation.py:887]: Would this ever explicitly raise an `IndexError` ? I thought `params[-1]` would always return the last item in the list? When would we cause the `IndexError`

**Review by Jaybsoni — APPROVED** [review]
Looks pretty straight forward 👍🏼

File: pennylane/operation.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/operation.py:887]: If nothing at all is passed.  For example, someone tries `qml.S()`. Just copied over this behavior from the initialization of `Observable`.

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -12,6 +12,17 @@
...
+* Wires can be passed as the final argument to an `Operator`, instead of requiring
+  the wires to be explicitly specified with keyword `wires`. This functionality already
+  existed for `Observable`'s, but now extends to all `Operator`'s.
+  [(#2432)](https://github.com/PennyLaneAI/pennylane/pull/2432)
+
+  ```pycon
+  >>> qml.S(0)
+  S(wires=[0])
+  >>> qml.CNOT((0,1))
+  CNOT(wires=[0, 1])
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:24]:
```suggestion
  CNOT(wires=[0, 1])
  ```
```

**Comment by antalszava** [comment]
The Codecov failure seems to be a false negative, overriding the check.

**Pull request merged**